Create array.with().md (#2854)
Codecademy/docs: array.with().md

> ---

Reviewer: Unfortunately, the issue for this entry was missing some information. You should create a new folder and rename the file to match the following path:
content/javascript/concepts/terms/arrays/terms/with/with.md

Author: I'm not sure what this means, could you please clarify?

Reviewer: This entry has the incorrect name, and it is not saved in the appropriate location. 

The name of the file should be "with". There should not be parentheses in the file name.

The file should be saved in a folder you create named "with". That folder should be saved at -
content/javascript/concepts/terms/arrays/terms/

>  Title: 'array.prototype.with()'

Reviewer: ```suggestion
 Title: '.with()'
```

> Description: 'This teaches JavaScript developers how to use the array method .with()'

Reviewer: ```suggestion
Description: 'Returns a new updated array with the given value at the specified index.'
```

Subjects:
  - 'Web development'
Tags:
  - 'Arrays'
  - 'Methods'
CatalogContent:
  -'course/introduction-to-javascript'

>   -'paths/front-end-engineer-career-path'

Reviewer: ```suggestion
  - 'paths/front-end-engineer-career-path'
```

---

> ## .with()

Reviewer: ```suggestion
```

> The `.with()` **method** modifies the value of an element in an **array**. Since `.with()` is a **copying method**, it creates a new array. 

Reviewer: ```suggestion
The **`.with()`** method modifies the value of an element in an array. Since `.with()` is a copying method, it creates a new array. 
```

Reviewer: Move this up to line 14

> ## Syntax:

Reviewer: ```suggestion
## Syntax
```

> >`arrayName.with(index, value);`

Reviewer: ```suggestion

```pseudo
arrayName.with(index, value);
```
```

> `.with()` takes the following parameters: 

Reviewer: add an empty line after this

> 1. The index of the array element

Reviewer: ```suggestion
- `index`: The location of the array element to be altered.
```

> 2. The new value for the array element.

Reviewer: ```suggestion
- `value`: The updated value for the array element.
```

> ## Example:

Reviewer: ```suggestion
## Example
```

Create and modify an array of favorite colors. 

```js
let favoriteColors = [‘blue’, ‘red’, ‘brown’, ‘grey’];

> // change red to purple

Reviewer: ```suggestion
// Updating red to purple
```

favoriteColors.with(1, ‘purple’);

console.log(favoriteColors);

> //
> Prints 'blue, red, brow, grey'. 

Reviewer: ```suggestion
// Prints 'blue, red, brow, grey'. 
```

```

> Notice that `.with()` should change 'red' to 'purple'. But `console.log(favouriteColors)` still prints red. The reason is that `.with()` does not modify an array, it returns a new array.

Reviewer: ```suggestion
> **Note:** `.with()` was used to change `'red'` to `'purple'` in the example above. However, `console.log(favouriteColors)` still prints `red`. The reason is that `.with()` does not modify an array, it returns a new array.
```

```js
// Create an array of favorite colors.  

const favoriteColors = [‘blue’, ‘red’, ‘brown’, ‘grey’];

console.log(favoriteColors.with(1, ‘purple’);

// Prints blue, purple, brown, grey. 

let favoriteColorsModified = favoriteColors.with(1, ‘purple’);

console.log(favoriteColorsModified);

// Prints 'blue, purple, brown, grey'. 
```

> ## Codebyte example: 

Reviewer: Codecademy docs infrastructure doesn't currently support this method. So, this example will result in an error. Please delete this example
